sdk/go/pulumi/resource_test.go

```diff
@@ -1246,6 +1246,12 @@ func newTestRes(t *testing.T, ctx *Context, name string, opts ...ResourceOption)
 	return &res
 }
 
+func newUnknwonRes() *testRes {
```

Comment:
Not sure if there's a better way to do this. Is there a way to make the mocks behave like a preview?